src/python/library/tritonclient/utils/cuda_shared_memory/__init__.py

```diff
@@ -25,117 +25,59 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-import ctypes
-import os
-import struct
-from ctypes import *
+# Check for dependency before other import so other imports can assume
+# the module is available (drop "try ... except .."")
+try:
+    from cuda import cudart
+except ModuleNotFoundError as error:
+    raise RuntimeError(
+        'CUDA shared memory utilities require Python package "cuda-python" to be installed'
+    ) from error
```

Comment:
If it is always an requirement, it will fail the client installation if `cuda-python` is not in PyPi registry. I have seen it failed for this reason in ubuntu 18.04. I think there will always be scenario that its supporting platforms are not in sync with previous style (provide our util C library). But I think using official package for Python is more portable and easier for development than maintaining our own.

I can look into making some packages require for installation but okay to fail the installation, which should make the change less visible to user. From what I gather is that this is not feasible to just use `setup.py` syntax and will need some additional post-installation script to do so.

Reply:
Unfortunately the post-installation script won't work for package fetched from PyPi, so I think we have to make it a general dependency which will fail the installation if `cuda-python` can't be installed. I don't think it is an issue as `cuda-python` has been available since Python 3.7 and we can probably assume that user will have Python >= 3.7. @Tabrizian how does it sound?